secrethub_wrapper.go

```diff
@@ -24,37 +24,61 @@ struct SecretVersion {
 	time CreatedAt;
 	char* Status;
 };
+
+struct Client {
+	int ID;
+};
 #include <stdbool.h>
+#include <stdlib.h>
 */
 import "C"
 
 import (
 	"encoding/json"
+	"github.com/secrethub/secrethub-go/pkg/secrethub"
 	"os"
 	"strings"
-
-	"github.com/secrethub/secrethub-go/pkg/secrethub"
+	"unsafe"
 )
 
-// Client creates and returns a new Go client to be used by xgo clients.
-func Client() (*secrethub.Client, error) {
+var clients = make(map[int]secrethub.ClientInterface)
+var nextClientID = 1;
+
+//export new_Client
+func new_Client(errMessage **C.char) *C.struct_Client{
 	options := []secrethub.ClientOption{
 		secrethub.WithAppInfo(&secrethub.AppInfo{
 			Name:    "secrethub-xgo",
 			Version: "0.1.0",
 		}),
 	}
-	return secrethub.NewClient(options...)
-}
-
-// Read retrieves a secret by its path.
-//export Read
-func Read(path *C.char, errMessage **C.char) C.struct_SecretVersion {
-	client, err := Client()
+	client, err := secrethub.NewClient(options...)
 	if err != nil {
 		*errMessage = C.CString(err.Error())
-		return C.struct_SecretVersion{}
+		return nil
 	}
+	cClient := (*C.struct_Client)(C.malloc(C.size_t(unsafe.Sizeof(C.struct_Client{}))))
+	cClient.ID = C.int(nextClientID)
+	clients[nextClientID] = client
+	nextClientID++
+	return cClient
```

Comment:
What happens if this is called from multiple threads? Should we guard against that?